etna/models/base.py

```diff
@@ -631,7 +631,9 @@ def forecast(self, ts: "TSDataset", prediction_size: int) -> "TSDataset":
         predictions = self.raw_predict(test_dataset)
         future_ts = ts.tsdataset_idx_slice(start_idx=self.encoder_length, end_idx=self.encoder_length + prediction_size)
         for (segment, feature_nm), value in predictions.items():
-            future_ts.df.loc[:, pd.IndexSlice[segment, feature_nm]] = value[:prediction_size, :]
+            # we don't want to change dtype after assignment, but there can happen cast to float32
+            dtype = future_ts.df.loc[:, pd.IndexSlice[segment, feature_nm]]
+            future_ts.df.loc[:, pd.IndexSlice[segment, feature_nm]] = value[:prediction_size, :].astype(dtype)
```

Comment:
May be we should cast to float64 instead if we suppose it

Reply:
I just think that using current dtype will be more safe in the future.